docs/source/docs/configuration.md

```diff
@@ -2,19 +2,459 @@
 
 Besides the arguments provided to the command line tools, detailed training and
 prediction configuration is specified in a `yaml` file. The configuration file contains
-two sections, `dataset` and `train`.
+two sections, `dataset` and `train`. Within `dataset` one specifies the data types,
+sizes, as well as the data loader to use. The `train` section specifies parameters
+related to the neural network. `yaml` files are used as they can be easily generated
+from basic Python dictionary structures.
 
 ## Dataset section
 
+The `dataset` section specifies the path to the data to be used during training, the
+data loader to use as well as the specific arguments to configure the data loader.
+
+### Dir key - Required
+
+The paths to the training, validation and testing data are specified under a `dir`
+dictionary key like this:
+
+```yaml
+dataset:
+  dir:
+    train: "data/test/h5/paired/train" # folder contains training data
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
+```
+
+Multiple dataset directories can be specified, such that data is sampled across several
+folders:
+
+```yaml
+dataset:
+  dir:
+    train:
+      - "data/test/h5/paired/train1"
+      - "data/test/h5/paired/train2" # folders contains training data
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
+```
+
+### Format key - Required
+
+The type of data we supply the data loaders will influence the behaviour, so we must
+specify the data type using the `format` key. Currently, DeepReg data loaders support
+nifti and h5 file types - alternate file formats will raise errors in the data loaders.
+To indicate which format to use, pass a string to this field as either "nifti" or "h5":
+
+```yaml
+dataset:
+  dir:
+    train: "data/test/h5/paired/train1" # folders contains training data
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
+  format: "nifti"
+```
+
+Depending on the data type, DeepReg will expect the images and labels to be stored in
+specific structures: check the [data loader configuration](dataset_loader.html) for more
+details.
+
+### Labeled key - Required
+
+The `labeled` key indicates whether labels should be used during training. A Boolean is
+used to indicate the usage of labels:
```

Comment:
boolean values are case sensitive, we recommend using only `true` and `false`, all lower case

https://pyyaml.org/wiki/PyYAMLDocumentation